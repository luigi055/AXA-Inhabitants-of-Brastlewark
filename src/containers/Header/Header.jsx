// @flow
import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import * as actions from "./../../redux/actions/actions";
import AutoCompleteInput from "./../../components/AutoCompleteInput/AutoCompleteInput";
import HeaderComponent from "./HeaderStyled";

type Props = {
  gnomes: Array<string>,
  getSearchText: Function
};

class Header extends Component<Props> {
  updateSearchTerm = (term: string) => this.props.getSearchText(term);

  render() {
    const gnomes = this.props.gnomes.map(gnome => gnome.name);
    return (
      <HeaderComponent>
        <div className="row">
          <h1>
            <Link to="/" href="/">
              Gnome
            </Link>
          </h1>
          <AutoCompleteInput
            stateName="searchTerm"
            parentUpdateState={this.updateSearchTerm}
            autoCompleteItems={gnomes}
            maxSuggests={5}
            maxWidth="400px"
            labelName="Search Gnome: "
            placeholder="Type a Gnome Name"
            includeSearchTerm
          />
        </div>
      </HeaderComponent>
    );
  }
}

function mapStateToProps(state) {
  return {
    gnomes: state.gnomes,
    searchTerm: state.searchTerm
  };
}

export default connect(mapStateToProps, actions)(Header);
