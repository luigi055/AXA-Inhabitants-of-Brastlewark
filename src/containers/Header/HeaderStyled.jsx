import styled from "styled-components";

const HeaderComponent = styled.header`
  align-items: center;
  background: ${({ theme }) => theme.headerBG};
  box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.5);
  display: flex;
  height: 50px;
  justify-content: center;

  h1 {
    a {
      color: ${({ theme }) => theme.logoTextColor};
      text-decoration: none;
    }
  }

  div.row {
    align-items: center;
    display: flex;
    justify-content: center;
    padding: 0 5px;
    max-width: 900px;
    width: 100%;
  }
`;

export default HeaderComponent;
