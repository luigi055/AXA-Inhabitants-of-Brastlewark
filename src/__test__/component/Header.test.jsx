import React from "react";
import { shallow } from "enzyme";
import Header from "./../../containers/Header/Header";
import { Link } from "react-router-dom";

// Testing Header for get acquainted with Enzyme
// Having some problems with jest since it can't process
// Images i have to improve the configuration
describe("Test Header", () => {
  it("should render Header correctly", () => {
    const Wrapper = shallow(<Header />);
    expect(Wrapper).toMatchSnapshot();
  });

  it("should render Header with Head Link BrastlewarkerS", () => {
    const Wrapper = shallow(<Header to="/" href="/" />);
    expect(
      Wrapper.containsMatchingElement(
        <Link to="/" href="/">
          BrastlewarkerS
        </Link>
      )
    ).toBe(true);
  });
});
