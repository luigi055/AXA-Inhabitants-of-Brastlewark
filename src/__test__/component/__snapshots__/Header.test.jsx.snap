// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header should render Header correctly 1`] = `
<styled.header>
  <div
    className="row"
  >
    <h1>
      <Link
        href="/"
        replace={false}
        to="/"
      >
        BrastlewarkerS
      </Link>
    </h1>
  </div>
</styled.header>
`;
